swarm/network/stream/stream.go

```diff
@@ -93,14 +94,21 @@ func NewRegistry(localID enode.ID, delivery *Delivery, syncChunkStore storage.Sy
 	}
 	streamer.api = NewAPI(streamer)
 	delivery.getPeer = streamer.getPeer
-	streamer.RegisterServerFunc(swarmChunkServerStreamName, func(_ *Peer, _ string, _ bool) (Server, error) {
-		return NewSwarmChunkServer(delivery.chunkStore), nil
-	})
+
+	if options.DoServeRetrieve {
+		streamer.RegisterServerFunc(swarmChunkServerStreamName, func(_ *Peer, _ string, _ bool) (Server, error) {
+			return NewSwarmChunkServer(delivery.chunkStore), nil
```

Comment:
aren't we possibly swallowing an error here?

Reply:
I am not sure, to me it seems that the `streamer.RegisterServerFunc` does not return with error and all it does is to register a function to a map, guarded with locks.

Reply:
👍